packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -110,6 +111,14 @@ Rect _globalBoundingBoxFor(BuildContext context) {
 /// B to A, route A's hero's widget is, by default, placed over where route B's
 /// hero's widget was, and then the animation goes the other way.
 ///
+/// ### Nested Navigators
+///
+/// If either or both routes contain nested [Navigator]s, only [Hero]s
```

Comment:
If there aren't any nested navigators we still only consider heroes from the topmost route?

Reply:
The important bit is that in the nested navigator case, its the topmost route **of the nested navigator** (and not the topmost route of the navigator controlling the hero transition)

Reply:
I rephrased this a bit and emphasized *nested navigators*.